tests/0143-exponential_backoff_mock.c

```diff
@@ -304,15 +307,20 @@ static void test_heartbeat_find_coordinator(rd_kafka_mock_cluster_t *mcluster,
                                             rd_kafka_conf_t *conf) {
         rd_kafka_t *consumer;
         rd_kafka_message_t *rkm;
+        int ApiKey;
         SUB_TEST();
         test_conf_set(conf, "auto.offset.reset", "earliest");
         test_conf_set(conf, "enable.auto.commit", "false");
 
         consumer = test_create_consumer(topic, NULL, conf, NULL);
 
+        if (test_consumer_group_protocol_classic()) {
+                ApiKey = RD_KAFKAP_Heartbeat;
+        } else {
+                ApiKey = RD_KAFKAP_ConsumerGroupHeartbeat;
+        }
```

Comment:
This should be opposite condition. if condition should be for `consumer` as that is the new one.

Reply:
I've always used always `test_consumer_group_protocol_classic()`, even if implementing the other function (that I can remove) so it's consistent for search, also it isolates the parts that we want to remove in a distant future.
Also if there was a third protocol, we hope not, that should start from the assumptions and tests of the consumer protocol, not with those of the classic protocol.

Reply:
I think we should add a small point on changing this in a major version. Can you add that so that we don't forget?

Reply:
We should be using `test_consumer_group_protocol_consumer` at all the places instead of `test_consumer_group_protocol_classic`. Reason behind this is:

1. Let's say we add a new protocol in future. We want all the tests to run. We are making sure of that by using `!test_consumer_group_protocol_consumer()` here instead.
2. In `!test_consumer_group_protocol_classic()` condition, we will skip those tests with the new protocol added.

Reply:
N,o in this case we're running the `RD_KAFKAP_ConsumerGroupHeartbeat` test that is more appropriate than running the classic one, if we use `test_consumer_group_protocol_consumer` ... `else` in case of a new protocol we run the classic tests, while the newer ones would be more appropriate.

Reply:
I am talking about all the places. Not just here. Basically the idea is to make sure that we run all the test cases when we add new protocols. Let's ensure that when there is a new protocol, all the tests are executed.

Reply:
yes, all tests about the new protocol, the tests about the classic protocol don't have to be executed as they won't be applicable. Probably the tests about the 848 protocol won't be applicable either but it's more probable it'll be similar to 848 in case it happens.